tests/test_ble_lifecycle_receive_targets.py

```
@@ -665,259 +727,652 @@ def join(timeout: float | None = None) -> None:
-        BLEReceiveRecoveryService._coordinator_wait_for_event(
-            SimpleNamespace(),
-            "evt",
-            timeout=None,
+        assert controller._as_usable_callable(None) is None
+        assert controller._is_unusable_mock_value(None) is True
+        assert controller._is_unusable_mock_value("ready") is False
+
+        iface._get_lifecycle_controller = MagicMock()
+        assert controller._get_lifecycle_controller() is None
+
+        iface._get_lifecycle_controller = lambda: MagicMock()
```

Ruff (0.15.6)

[warning] Lambda may be unnecessary; consider inlining inner function

Inline function call

(PLW0108)